### PR TITLE
rustup https://github.com/rust-lang/rust/pull/60417/

### PR DESCRIPTION
**Pull request by matthiaskrgr**
cc https://github.com/rust-lang/rust/issues/60448

changelog: none

### CODE REVIEW

**Comment by matthiaskrgr** [comment]
looks spurious
````
   Compiling reqwest v0.9.16
    Finished dev [unoptimized + debuginfo] target(s) in 3m 48s
  Installing /Users/travis/.cargo/bin/rustup-toolchain-install-master
   Installed package `rustup-toolchain-install-master v1.6.0` (executable `rustup-toolchain-install-master`)
detecting the channel of the `9b67bd42b7cbf97f72d039afcba02f5177d0d68c` toolchain...
error: https://s3-us-west-1.amazonaws.com/rust-lang-ci2/rustc-builds/9b67bd42b7cbf97f72d039afcba02f5177d0d68c/rust-src-nightly.tar.xz: timed out
error: toolchain 'master' is not installed
info: backtrace:

````

File: clippy_lints/src/utils/author.rs [hunk 1]
```diff
@@ -495,7 +495,7 @@ impl<'tcx> Visitor<'tcx> for PrintVisitor {
             ExprKind::Err => {
                 println!("Err = {}", current);
             },
-            ExprKind::Use(ref expr) => {
+            ExprKind::DropTemps(ref expr) => {
                 let expr_pat = self.next("expr");
                 println!("Use(ref {}) = {};", expr_pat, current);
```
Comment (eddyb) [inline clippy_lints/src/utils/author.rs]: This is wrong, should say `DropTemps`.

File: clippy_lints/src/utils/hir_utils.rs [hunk 2]
```diff
@@ -607,8 +607,8 @@ impl<'a, 'tcx: 'a> SpanlessHash<'a, 'tcx> {
                 }
             },
             ExprKind::Err => {},
-            ExprKind::Use(ref e) => {
-                let c: fn(_) -> _ = ExprKind::Use;
+            ExprKind::DropTemps(ref e) => {
+                let c: fn(_) -> _ = ExprKind::DropTemps;
                 c.hash(&mut self.s);
```
Comment (eddyb) [inline clippy_lints/src/utils/hir_utils.rs]: Huh, this should hash `mem::discriminant` (or thereabouts) like rustc does.
Preexisting though, better not to fix it in this PR.
cc @oli-obk @Manishearth

File: clippy_lints/src/utils/inspector.rs [hunk 3]
```diff
@@ -330,7 +330,7 @@ fn print_expr(cx: &LateContext<'_, '_>, expr: &hir::Expr, indent: usize) {
         hir::ExprKind::Err => {
             println!("{}Err", ind);
         },
-        hir::ExprKind::Use(ref e) => {
+        hir::ExprKind::DropTemps(ref e) => {
             println!("{}Use", ind);
```
Comment (eddyb) [inline clippy_lints/src/utils/inspector.rs]: This should say `DropTemps`.

**Comment by eddyb** [comment]
Not sure if this will fail/be stopped, but I left a couple comments with some necessary changes.

**Comment by matthiaskrgr** [comment]
https://github.com/rust-lang/rust-clippy/pull/4056

**Comment by Manishearth** [comment]
#4056

could have just pushed to this one :smile:

**Pull request closed without merging**